app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,291 @@
+                lastName = lastName,
+                location = location,
+                dateOfBirth = date,
+            )
+            val validMail = isValidEmail(email)
+            val validLoc = isValidLocation(location)
+            val validDob = isValidDateOfBirth(date)
+            if (validMail && validLoc && validDob && sqlDate != null) {
+                sqlDate?.let { sqlDate ->
+                    val outcome = createUserUseCase(
+                        firstName = firstName,
+                        lastName = lastName,
+                        email = email,
+                        date = sqlDate,
+                        location = location
+                    )
+                    Log.e(VIEWMODEL_ERROR_TAG, "${outcome}")
+                    when (outcome) {
+                        CreateUserOutcome.CREATE_USER_FAILURE -> {
+                            createAccountState.value =
+                                CreateAccountUiState.Idle(
+                                    email = email,
+                                    firstName = firstName,
+                                    lastName = lastName,
+                                    location = location,
+                                    dateOfBirth = date,
+                                )
+                        }
+                        CreateUserOutcome.CREATE_USER_DUPLICATE -> {
+                            CreateAccountUiState.Idle(
+                                email = email,
+                                firstName = firstName,
+                                lastName = lastName,
+                                location = location,
+                                dateOfBirth = date,
+                                isDuplicateUser = true,
+                            )
+                        }
+                        CreateUserOutcome.CREATE_USER_SUCCESS -> {
+                            createAccountState.value = CreateAccountUiState.Loaded
+                        }
+                    }
+                }
+            } else {
+                createAccountState.value = CreateAccountUiState.Idle(
+                    isEmailError = validMail,
+                    isLocationError = validLoc,
+                    isDateError = validDob,
+                    email = email,
+                    firstName = firstName,
+                    lastName = lastName,
+                    location = location,
+                    dateOfBirth = date,
+                )
+            }
+        }
+        Log.e(VIEWMODEL_ERROR_TAG, "${createAccountState.value}")
+    }
+
+    private fun isValidLocation(location: String): Boolean {
+        return location.length <= LOCATION_CHAR_LIMIT
+    }
+
+    private fun isValidEmail(email: String): Boolean {
+        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    private fun isValidDateOfBirth(dob: String): Boolean {
+        dob.forEach {
+            if (!it.isDigit()) {
```

Comment:
Where do you filter slashes and dashes?

Reply:
I now filter for slashes and dashes first.